pkg/apis/core/helper/helpers.go

```diff
@@ -20,12 +20,12 @@ import (
 
 // GetCondition returns the condition with the given <conditionType> out of the list of <conditions>.
 // In case the required type could not be found, it returns nil.
-func GetCondition(conditions []core.Condition, conditionType core.ConditionType) *core.Condition {
-	for _, condition := range conditions {
+func GetCondition(conditions []core.Condition, conditionType core.ConditionType) (int, *core.Condition) {
```

Comment:
Instead of extending this, could you rather create a new method `GetConditionIndex` and reuse the index in `GetCondition`?